score.go

```diff
@@ -127,14 +130,24 @@ func (d *indexData) scoreFileUsingBM25(fileMatch *FileMatch, doc uint32, cands [
 	termFreqs := map[string]int{}
 	for _, cand := range cands {
 		term := string(cand.substrLowered)
-		termFreqs[term]++
+
+		if cand.fileName {
```

Comment:
In text search, it's super common to "boost" matches on document title. The same thing applies here, as a filename match is strong signal. The choice of 2 is quite conservative, and means that a match on filename counts for two matches on content. I tried a few choices and didn't see evidence that a larger number would help.

Reply:
Neat!